libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -231,6 +232,51 @@ bool StaticAnalyzer::visit(InlineAssembly const& _inlineAssembly)
 	return true;
 }
 
+bool StaticAnalyzer::visit(BinaryOperation const& _operation)
+{
+	if (
+		_operation.rightExpression().annotation().isConstant &&
```

Comment:
I think this can be safely change to `isPure` and then it should also catch cases like `uint constant a = -1; uint constant b = 7 / (a+1);`